app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/TagsManager.java

```diff
@@ -40,167 +47,198 @@ boolean editTag(Tag tag) {
     }
 
     boolean clearTag(Tag tag) {
-        if (tag.getKey().equals(TAG_GPS_TIMESTAMP)) {
-            tag.setValue("00:00:00");
-        } else {
-            switch (tag.getObjectType()) {
-                case STRING:
-                    tag.setValue("");
-                    break;
-                case INTEGER:
-                case DOUBLE:
-                    tag.setValue("0");
-                default:
-            }
+        switch (tag.getTagType().getInputType()) {
+            case TEXT_STRING:
+                tag.setValue("");
+                break;
+            case TIMESTAMP_STRING:
+                tag.setValue("00:00:00");
+                break;
+            case DATETIME_STRING:
+                tag.setValue("2001-01-01 00:00:00");
+                break;
+            case DATE_STRING:
+                tag.setValue("2001-01-01");
+                break;
+            case RANGED_STRING:
+                tag.setValue("0.00000");
```

Comment:
We are using it for lat, lng REF, so user can enter one of the following values: E, W, S, N.  Does this information change your mind about NO_VALUE in  RANGE_STRING case?